llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -6746,6 +6748,122 @@ SDValue SITargetLowering::lowerFLDEXP(SDValue Op, SelectionDAG &DAG) const {
   return DAG.getNode(ISD::FLDEXP, DL, VT, Op.getOperand(0), TruncExp);
 }
 
+static unsigned getExtOpcodeForPromotedOp(SDValue Op) {
```

Comment:
I would hope there would be a generic helper for this somewhere

Reply:
I haven't found one, there are some `getExtendFor` functions but they're not for binops